mat64/dense.go

```diff
@@ -427,3 +429,63 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	var err error
+	buf := new(bytes.Buffer)
+	err = writeBin(buf, int64(m.mat.Rows))
+	if err != nil {
+		return nil, err
+	}
+
+	err = writeBin(buf, int64(m.mat.Cols))
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < m.mat.Rows; i++ {
+		// We don't write out the length of the slice as
```

Comment:
I found this comment confusing. I would prefer you get rid of this, and before the for loop write
````
// Write all of the elements in condensed form  
````
or something like that.